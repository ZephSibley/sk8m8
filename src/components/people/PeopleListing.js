import React, { useState, useEffect } from 'react';
import List from '@material-ui/core/List';
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer';

import spinner from '../../assets/img/ajax-loader.gif';
import PersonListItem from '../../components/people/PersonListItem';
import ChatView from '../../components/people/ChatView';


const PeopleListing = props => {
    // Prop: requests; http client
    // Prop: location; latlong array

    const [peopleListInfo, setPeopleListInfo] = useState([{
        key: 0,
        username: 'Loading...',
        avatar: spinner,
        status: '',
        chat: function(){}
    }]);
    const [interlocutor, setInterlocutor] = useState('');
    const [chatting, setChatting] = useState(false);

    const toggleChat = username => {
        setChatting(x => !x)
        setInterlocutor(username || '')
    }

    useEffect(() => {
        // Get location, add to queryurl
        if (props.location[0] !== null && props.location[1] !== null) {
            props.requests.get(
                `${process.env.REACT_APP_ENDPOINT}People/Find?latitude=${props.location[0]}&longitude=${props.location[1]}&radius=${1}`,
                {
                    headers: {
                        'Authorization': `Bearer ${localStorage.getItem('token')}`
                    }
                }
            ).then(response => {
                response.data.chat = toggleChat
                setPeopleListInfo(response.data)
            }).catch(err =>
                setPeopleListInfo([{
                    key: 0,
                    username: 'Something went wrong D:',
                    avatar: '',
                    status: err.message,
                    chat: function(){}
                }])
            );
        }        
    }, [props.requests, props.location])

    return (
        <React.Fragment>
            <List>
                {peopleListInfo.map(PersonListItem)}
            </List>
            <div style={{marginBottom: 100}}></div>
            <SwipeableDrawer
                open={chatting}
                onClose={() => setChatting()}
                onOpen={() => setChatting()}
            >
                <ChatView interlocutor={interlocutor} />
            </SwipeableDrawer>
        </React.Fragment>
    )
}

export default PeopleListing