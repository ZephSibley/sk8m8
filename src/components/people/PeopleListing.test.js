import React from 'react';
import renderer from 'react-test-renderer';
import mockAxios from '../../utils/__mocks__/mockAxios';
import PeopleListing from './PeopleListing';

test('Basic Rendering', () => {
    mockAxios.get.mockImplementationOnce(() =>
        Promise.resolve([
            {
                username: 'Name!',
                avatar: '../../assets/img/ajax-loader.gif',
                bio: 'hey hey hey',
                chat: function(){}
            },
            {
                username: 'testymctestface',
                avatar: '../../assets/img/ajax-loader.gif',
                bio: 'amwd;uneagjnegiohwedjwnfun',
                chat: function(){}
            }
        ])
    )

    const component = renderer.create(
        <PeopleListing
            location={[50, 50]}
            requests={mockAxios}
        />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})