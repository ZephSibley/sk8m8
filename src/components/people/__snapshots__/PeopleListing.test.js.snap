// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
Array [
  <ul
    className="MuiList-root MuiList-padding"
  >
    <li
      className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
      disabled={false}
    >
      <div
        className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
      >
        <div
          className="MuiAvatar-root"
        >
          <img
            alt="Loading..."
            className="MuiAvatar-img"
            src={<ForwardRef(WithStyles) />}
          />
        </div>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
        >
          Loading...
        </span>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
        >
          
        </p>
      </div>
    </li>
    <li
      className="MuiDivider-root MuiDivider-inset"
      role="separator"
    />
  </ul>,
  <div
    style={
      Object {
        "marginBottom": 100,
      }
    }
  />,
]
`;
